src/ralph/models/edx/browser.py

```diff
@@ -0,0 +1,49 @@
+"""Browser event model definitions"""
+
+from pathlib import Path
+from typing import Literal, Union
+
+from pydantic import AnyUrl, constr
+
+from .base import BaseEventModel
+
+MD5_HASH_LEN = 32
+
+
+class BaseBrowserEventModel(BaseEventModel):
+    """Represents the base browser event model all browser events inherit from.
+
+    This type of events is triggered on (XHR) POST/GET request to the `/event` URL.
+
+    Attributes:
+        event_source (str): Consists of the value `browser`.
+        page (Path): Consists of the URL (with hostname) of the visited page.
+            Retrieved with:
+                `window.location.href` from the JavaScript front-end.
+        session (str): Consists of the md5 encrypted Django session key or an empty string
+    """
+
+    event_source: Literal["browser"]
+    page: Union[AnyUrl, Path]
+    session: Union[
+        constr(min_length=MD5_HASH_LEN, max_length=MD5_HASH_LEN), Literal[""]
```

Comment:
I would use a regular expression instead:

```suggestion
        constr(regex=r'^[a-f0-9]{32}$'), Literal[""]
```

(and remove the `MD5_HASH_LEN` definition).

Reply:
Good Idea) Thanks)